### PR TITLE
DOC: Simplify contributing intro

### PR DESCRIPTION
**Pull request by mroeschke**
* Reduced some preamble
* Updated some links
* Use `code-block:: shell` where appropriate

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
nice! @noatamir fancy taking a look? interested in adding yourself to https://github.com/pandas-dev/pandas/blob/main/.github/CODEOWNERS for `doc/source/development` (and anything else)?

**Comment by noatamir** [comment]
Hmm, @mroeschke the doc build is failing and it looks like an extensive change. Any chance you have screenshots from a local build or can you please fix the build issue?

**Comment by mroeschke** [comment]
@noatamir was able to fix the issue and generate the preview

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -138,18 +95,20 @@ Below are some useful resources for learning more about forking and pull request
...
-`GitHub has instructions <https://help.github.com/set-up-git-redirect>`__ for installing git,
+`GitHub has instructions <https://docs.github.com/en/get-started/quickstart/set-up-git>`__ for installing git,
 setting up your SSH key, and configuring git.  All these steps need to be completed before
 you can work seamlessly between your local repository and GitHub.
 
 .. _contributing.forking:
 
-Forking
--------
+Forking pandas
```
Comment (noatamir) [inline doc/source/development/contributing.rst:104]:
```suggestion
Create a fork of pandas
```

File: doc/source/development/contributing.rst [hunk 2]
```diff
@@ -138,18 +95,20 @@ Below are some useful resources for learning more about forking and pull request
...
+--------------
```
Comment (noatamir) [inline doc/source/development/contributing.rst:105]:
```suggestion
-----------------------
```

File: doc/source/development/contributing.rst [hunk 3]
```diff
@@ -138,18 +95,20 @@ Below are some useful resources for learning more about forking and pull request
...
 
 You will need your own fork to work on the code. Go to the `pandas project
```
Comment (noatamir) [inline doc/source/development/contributing.rst:107]:
```suggestion
You will need your own copy of pandas (aka fork) to work on the code. Go to the `pandas project
```

File: doc/source/development/contributing.rst [hunk 4]
```diff
@@ -138,18 +95,20 @@ Below are some useful resources for learning more about forking and pull request
...
 page <https://github.com/pandas-dev/pandas>`_ and hit the ``Fork`` button. You will
```
Comment (noatamir) [inline doc/source/development/contributing.rst:108]:
```suggestion
page <https://github.com/pandas-dev/pandas>`_ and hit the ``Fork`` button. Please uncheck the box to copy only the main branch before selecting ``Create Fork``.
```

File: doc/source/development/contributing.rst [hunk 5]
```diff
@@ -138,18 +95,20 @@ Below are some useful resources for learning more about forking and pull request
...
-want to clone your fork to your machine::
+want to clone your fork to your machine
```
Comment (noatamir) [inline doc/source/development/contributing.rst:109]:
```suggestion
You will want to clone your fork to your machine
```

**Comment by noatamir** [comment]
This is great! I made a few small suggestions.

**Comment by mroeschke** [comment]
Thanks all the suggestions looked great

**Review by noatamir — APPROVED** [review]

**Pull request merged**